src/test/java/redis/clients/jedis/authentication/TokenBasedAuthenticationUnitTests.java

```diff
@@ -0,0 +1,342 @@
+package redis.clients.jedis.authentication;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.awaitility.Awaitility.await;
+import static org.awaitility.Durations.*;
+import static org.hamcrest.CoreMatchers.either;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+import java.util.Collections;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockedConstruction;
+
+import redis.clients.authentication.core.IdentityProvider;
+import redis.clients.authentication.core.IdentityProviderConfig;
+import redis.clients.authentication.core.SimpleToken;
+import redis.clients.authentication.core.Token;
+import redis.clients.authentication.core.TokenAuthConfig;
+import redis.clients.authentication.core.TokenListener;
+import redis.clients.authentication.core.TokenManager;
+import redis.clients.authentication.core.TokenManagerConfig;
+import redis.clients.authentication.core.TokenManagerConfig.RetryPolicy;
+import redis.clients.jedis.ConnectionPool;
+import redis.clients.jedis.EndpointConfig;
+import redis.clients.jedis.HostAndPort;
+
+public class TokenBasedAuthenticationUnitTests {
+
+  private HostAndPort hnp = new HostAndPort("localhost", 6379);
```

Comment:
You should use the `standalone0` endpoint instead of hardcoding the connection settings here.